lightly/models/utils.py

```diff
@@ -569,6 +570,176 @@ def random_prefix_mask(
+                device=device,
+            )
+        )
+    for _ in range(num_images_masked, B):
+        masks.append(torch.zeros((H, W), dtype=torch.bool, device=device))
+    # random.shuffle(masks)
+    for mask in masks:
+        print(mask.shape)
+    return torch.stack(masks)
+
+
+def random_block_mask_image(
+    size: Tuple[int, int],
+    num_mask: int,
+    min_num_mask_per_block: int = 4,
+    max_num_mask_per_block: Optional[int] = None,
+    min_aspect: float = 0.3,
+    max_aspect: Optional[float] = None,
+    max_attempts_per_block: int = 10,
+    device: Optional[Union[torch.device, str]] = None,
+) -> Tensor:
+    """Creates a random block mask for a single image.
+
+    Args:
+        size:
+            Size of the image for which to generate a mask.
+            Should be (height, width).
+        num_mask:
+            Number of patches to mask.
+        min_num_mask_per_block:
+            Minimum number of patches to mask per block.
+        max_num_mask_per_block:
+            Maximum number of patches to mask per block.
+        min_aspect:
+            Minimum aspect ratio of a masked block.
+        max_aspect:
+            Maximum aspect ratio of a masked block.
+        max_attempts_per_block:
+            Maximum number of attempts to find a valid block mask.
+        device:
+            Device on which to create the mask.
+    Returns:
+        A boolean tensor with shape (height, width) where each entry is True if the
+        patch should be masked and False otherwise.
+    """
+
+    if max_aspect is None:
+        max_aspect = 1 / min_aspect
+    if max_num_mask_per_block is None:
+        max_num_mask_per_block = num_mask
+
+    if max_num_mask_per_block < min_num_mask_per_block:
+        raise ValueError(
+            "max_num_mask_per_block must be greater or equal to min_num_mask_per_block."
+        )
+    if max_aspect < min_aspect:
+        raise ValueError("max_aspect must be greater or equal to min_aspect.")
+
+    log_min_aspect = math.log(min_aspect)
+    log_max_aspect = math.log(max_aspect)
+
+    H, W = size
+    mask = torch.zeros((H, W), dtype=torch.bool, device=device)
+    mask_count = 0
+    while mask_count < num_mask:
+        # Try masking a block
+        max_new_masked = min(num_mask - mask_count, max_num_mask_per_block)
+        delta = 0
+        for _ in range(max_attempts_per_block):
+            target_area = random.uniform(min_num_mask_per_block, max_new_masked)
+            aspect_ratio = math.exp(random.uniform(log_min_aspect, log_max_aspect))
+            h = int(round(math.sqrt(target_area * aspect_ratio)))
+            w = int(round(math.sqrt(target_area / aspect_ratio)))
+            if w < W and h < H:
+                top = random.randint(0, H - h)
+                left = random.randint(0, W - w)
+                num_masked = mask[top : top + h, left : left + w].sum().item()
+                new_masked = h * w - num_masked
```

Comment:
With type hints and other variable names, this could be clearer.
```suggestion
                num_already_masked: int = mask[top : top + h, left : left + w].sum().item()
               num_newly_masked: int = h * w - num_already_masked
```

Reply:
Didn't add the type hints as it makes it more verbose and IMO should be sufficiently clear that `sum().item()` returns an int.